### PR TITLE
EPICSYSTEM-225: Create staff modal form

### PR DESCRIPTION
**Pull request by nitheesh-aot**

### CODE REVIEW

File: compliance-web/src/components/App/Staff/StaffModal.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    ) => {
+      setFormData((prevValues) => ({ ...prevValues, [key]: newVal }));
+    };
+
+  const handleSubmit = (e: React.FormEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    // eslint-disable-next-line no-console
+    console.log(formData);
+    // if (user) {
+    //   updateUser({ ...user, ...formData });
```
Comment (dinesh-aot) [inline compliance-web/src/components/App/Staff/StaffModal.tsx:90]: remove these if not important

File: compliance-web/src/hooks/useStaff.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import { request } from "@/utils/axiosUtils";
+import { Position } from "@/models/Position";
+import { Permission } from "@/models/Permission";
+
+export interface MockUser {
+  id: number;
+  name: string;
+}
+
+const mockUsersList: MockUser[] = [
```
Comment (dinesh-aot) [inline compliance-web/src/hooks/useStaff.tsx:11]: If this is the mock for the dropdown list. You should call the Epic.Auth endpoint directly.
`
    {
        "id": "5bca57e4-1b39-498c-8872-732d3da4beaf",
        "first_name": "Shaelyn",
        "last_name": "Tolk",
        "email_address": "shaelyn.tolk@gov.bc.ca",
        "username": "0790439a8c0448ed94ebe98d759bce6a@idir"
    },`
You have to show the {first_name} {last_name} format and get the username as the selected value

File: compliance-web/src/utils/axiosUtils.ts [hunk 3]
```diff
@@ -1,42 +1,42 @@
...
-import axios, { AxiosError } from 'axios'
-import { User } from "oidc-client-ts"
+import { AppConfig, OidcConfig } from "@/utils/config";
+import axios, { AxiosError } from "axios";
+import { User } from "oidc-client-ts";
 
 export type OnErrorType = (error: AxiosError) => void;
 export type OnSuccessType = (data: any) => void;
 
 const client = axios.create({ baseURL: AppConfig.apiUrl });
```
Comment (dinesh-aot) [inline compliance-web/src/utils/axiosUtils.ts:9]: Shall we put a 3 times retry option with axios.

File: compliance-web/src/components/App/Staff/StaffModal.tsx (hunk 1 above)
Comment (nitheesh-aot) [reply compliance-web/src/components/App/Staff/StaffModal.tsx:90]: just kept from the scaffold one, need to implement the add and update in that area. 
I will remove that once it's implemented. It's already in my TODO list.

File: compliance-web/src/hooks/useStaff.tsx (hunk 2 above)
Comment (nitheesh-aot) [reply compliance-web/src/hooks/useStaff.tsx:11]: Yes, This for the name and other users dropdown list. I will connect to epic.auth for this.

File: compliance-web/src/utils/axiosUtils.ts (hunk 3 above)
Comment (nitheesh-aot) [reply compliance-web/src/utils/axiosUtils.ts:9]: Tanstack query is already doing that by default, I don't think we need to do it in Axios. we got more controls via TSQ

**Pull request merged**